src/main/java/racing/RacingGame.java

```diff
@@ -0,0 +1,54 @@
+package racing;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class RacingGame {
+
+    private final List<Car> cars;
+    private final RandomStrategy rs;
+
+    public RacingGame(List<Car> cars, RandomStrategy rs) {
+        this.cars = cars;
+        this.rs = rs;
+    }
+
+    public RacingGame(String carNames) {
+        this(carNames.split(","));
+    }
+
+    public RacingGame(String[] carNames) {
+        this(Arrays.stream(carNames)
+                .map(Car::new)
+                .collect(Collectors.toList()));
+    }
+    public RacingGame(List<Car> cars) {
+        this(cars, new RealRandomStrategy());
+    }
+
+    public List<Car> getCars() {
+        return cars;
+    }
+
+    public List<Car> getWinners() {
+        Map<Integer, List<Car>> groupByPosition = cars.stream().collect(Collectors.groupingBy(Car::getPosition));
+        int winnerPosition = getWinnerPosition(groupByPosition);
+        return getWinnerSortedByName(groupByPosition, winnerPosition);
+    }
+
+    private static List<Car> getWinnerSortedByName(Map<Integer, List<Car>> groupByPosition, int winnerPosition) {
+        List<Car> winners = groupByPosition.get(winnerPosition);
+        winners.sort(Comparator.comparing(Car::getName));
+        return winners;
+    }
+
+    private static int getWinnerPosition(Map<Integer, List<Car>> groupByPosition) {
+        ArrayList<Integer> positions = new ArrayList<>(groupByPosition.keySet());
```

Comment:
가급적 하위 구현체보다는 아래와 같이 상위의 인터페이스 타입인 List로 타입을 선언하시는 것이 좋습니다. 

```suggestion
        List<Integer> positions = new ArrayList<>(groupByPosition.keySet());
```

[[OOP] 우리는 왜 List list = new ArrayList(); 라고 쓰는가](https://www.itsmo.dev/interface-as-type/) 글을 참고해보시면 도움이될 것 같아요.

Reply:
상위 인터페이스 사용 참고하겠습니다~ 감사합니다~